src/time.cr

```diff
@@ -140,7 +140,15 @@ struct Time
     new(seconds: seconds + offset, nanoseconds: nanoseconds, kind: Kind::Local)
   end
 
-  def self.new(year, month, day, hour = 0, minute = 0, second = 0, nanosecond = 0, kind = Kind::Unspecified)
+  def self.new(year, month, day, hour = 0, minute = 0, second = 0, *, millisecond = 0, kind = Kind::Unspecified)
+    unless 0 <= millisecond <= 999_999
+      raise ArgumentError.new "Invalid time"
+    end
+
+    new(year, month, day, hour, minute, second, nanosecond: millisecond * 1_000, kind: kind)
```

Comment:
You should use the `NANOSECONDS_PER_MILLISECOND` constant. It even has the right number.

Reply:
Done